components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -78,682 +138,418 @@ function CustomMatchGroupInput._readDate(matchArgs)
 
-	-- Match is automatically marked finished upon page edit after a
-	-- certain amount of time (depending on whether the date is exact)
-	local threshold = match.dateexact and 30800 or 86400
-	match.finished = match.timestamp + threshold < NOW
-end
+		map.participants = MapFunctions.getParticipants(mapInput, opponents)
 
----@param match table
-function CustomMatchGroupInput._getTournamentVars(match)
-	match.cancelled = Logic.emptyOr(match.cancelled, Variables.varDefault('cancelled tournament', 'false'))
-	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
-	match.bestof = tonumber(Logic.emptyOr(match.bestof, Variables.varDefault('bestof')))
-	Variables.varDefine('bestof', match.bestof)
+		map.mode = MapFunctions.getMode(mapInput, map.participants, opponents)
 
-	MatchGroupInput.getCommonTournamentVars(match)
-end
+		Table.mergeInto(map.extradata, MapFunctions.getAdditionalExtraData(map, map.participants))
 
----@param match table
-function CustomMatchGroupInput._getLinks(match)
-	match.links = {}
-	for _, linkKey in pairs(LINKS_KEYS) do
-		match.links[linkKey] = match[linkKey]
-	end
-end
-
----@param match table
-function CustomMatchGroupInput._getExtraData(match)
-	match.extradata = {
-		casters = MatchGroupInput.readCasters(match),
-		ffa = 'false',
-	}
+		map.vod = Logic.emptyOr(mapInput.vod, match['vodgame' .. mapIndex])
 
-	for prefix, mapVeto in Table.iter.pairsByPrefix(match, 'veto') do
-		match.extradata[prefix] = (MapsData[mapVeto:lower()] or {}).name or mapVeto
-		match.extradata[prefix .. 'by'] = match[prefix .. 'by']
+		table.insert(maps, map)
+		match[mapKey] = nil
 	end
 
-	Table.mergeInto(match.extradata, Table.filterByKey(match, function(key, value)
-		return key:match('subgroup%d+header') end))
+	return maps
 end
 
----@param match table
-function CustomMatchGroupInput._adjustData(match)
-	--parse opponents + set base sumscores + set mode
-	CustomMatchGroupInput._opponentInput(match)
-
-	--main processing done here
-	local subGroupIndex = 0
-	for _, _, mapIndex in Table.iter.pairsByPrefix(match, 'map') do
-		subGroupIndex = CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
+---@param maps table[]
+---@param opponents table[]
+---@return fun(opponentIndex: integer): integer?
+function MatchFunctions.calculateMatchScore(maps, opponents)
+	return function(opponentIndex)
+		local calculatedScore = MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+		if not calculatedScore then return end
+		local opponent = opponents[opponentIndex]
+		return calculatedScore + (opponent.extradata.advantage or 0) - (opponent.extradata.penalty or 0)
 	end
+end
 
-	CustomMatchGroupInput._matchWinnerProcessing(match)
+---@param opponent table
+---@return table
+function MatchFunctions.getOpponentExtradata(opponent)
+	return {
+		advantage = tonumber(opponent.advantage),
+		penalty = tonumber(opponent.penalty),
+		score2 = opponent.score2,
```

Comment:
What's score2?

Reply:
need to check if needed
on sc2 there were old brackets that have a second set of scores in non grand final matches

Reply:
needed